src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_copy_b.cpp

```diff
@@ -239,7 +247,11 @@ void BrgemmCopyBKernel::init_brgemm_copy_b_kernel(
     matmul::brgemm_matmul_conf_t brgCopyKernelConf;
     brgCopyKernelConf.src_dt = conf.get_src_dt();
     brgCopyKernelConf.wei_dt = conf.get_wei_dt();
-    brgCopyKernelConf.orig_wei_dt = brgCopyKernelConf.wei_dt;
+    brgCopyKernelConf.orig_wei_dt = conf.get_original_wei_dt();
+    // WA: this hack is used to force f32->bf16 conversion (req_cvtps2bf16 flag in kernel constructor)
+    if (brgCopyKernelConf.orig_wei_dt != brgCopyKernelConf.wei_dt && brgCopyKernelConf.wei_dt == dnnl_bf16) {
+        brgCopyKernelConf.is_bf32 = true;
+    }
```

Comment:
I just noticed that `brgemm_matmul_matrix_B_reorder_t` [doesn't support](https://github.com/openvinotoolkit/oneDNN/blob/4500ee25b74566fe79616c54b8704d5c5a3c5fdf/src/cpu/x64/matmul/brgemm_matmul_reorders.cpp#L124) the case with `src_prc = bf16` and `wei_prc = fp32`. Due to this fact, we have to use hack with `bf32` here....
I'd say that this is one more argument to use `reorder` for weights repacking